src/config.rs

```diff
@@ -90,13 +89,29 @@ pub enum Email {
 /// ## Notes
 ///
 /// This function handles testing and non-testing environments.
-pub fn config_path() -> Result<PathBuf> {
+#[derive(Error, Debug)]
+pub enum ConfigError {
+    #[error("Failed to get home directory")]
+    HomeDir,
+    #[error("Failed to create config directory")]
+    CreateConfigDir,
+    #[error("Failed to create config file")]
+    CreateConfigFile,
+    #[error("Failed to save config file")]
+    SaveConfigFile,
```

Comment:
We need the TOML Error

```suggestion
    #[error(transparent)]
    TomlError(#[from] TomlError),  // above: use toml::ser::Error as TomlError;
    #[error("Failed to save config file")]
    SaveConfigFile,
```